Cython/Compiler/ExprNodes.py

```diff
@@ -1311,6 +1330,29 @@ def calculate_result_code(self):
     def generate_result_code(self, code):
         pass
 
+    @staticmethod
+    def for_type(pos, value, type):
```

Comment:
I'm slightly unclear what this means.

I'll add `calculate_constant_result()` to the value before returning it, but it's possible I've missed the point here.

Reply:
I meant to set the attribute (explicitly) when creating the node instance. Calling `calculate_constant_result()` is allowed to fail, so we should not use it outside of the `ConstantFolding` transform.

You could pass the `constant_result` also as an optional argument into this method, so that callers can decide if and how to calculate it. Default should probably be `constant_value_not_set`. Or calculate it here based on the type and value.

This is similar to `Parsing.wrap_compile_time_constant()`, BTW, although that has a slightly different focus. I'm not implying that there is room for reuse, though…